beacon-chain/core/transition/transition.go

```diff
@@ -257,16 +257,18 @@ func ProcessSlots(ctx context.Context, state state.BeaconState, slot primitives.
 					return nil, errors.Wrap(err, "could not process epoch with optimizations")
 				}
 			} else if state.Version() <= version.Deneb {
+				stateVersion := version.String(state.Version())
 				state, err = altair.ProcessEpoch(ctx, state)
```

Comment:
Can we drop the returned state? ProcessEpoch mutates the state argument, there is no benefit to conditionally returnign the state here. Refactor the method signature to only return an error. 

```suggestion
				err = altair.ProcessEpoch(ctx, state)
```